go-app.js

```diff
@@ -69,18 +68,104 @@ go.app = function() {
+                        name: 'states:retrieve-location',
+                        creator_opts : {
+                            full_geoid : choice.value,
+                            full_name : choice.label
+                        }
+                    };
                 }   
             });
         });
-      
+
+        self.states.add('states:retrieve-location', function(name, opts) {
+            return self
+                .http.get('http://wazimap.co.za/profiles/' + opts.full_geoid + '.json')
+                .then(function(response) {
+                    opts.data = response.data;
+                    return  self.states.create(
+                        'states:select-section', opts);
+                });
+        });
+
+        self.states.add('states:select-section', function(name, opts) {
+            return new ChoiceState(name, {
+                question: 'I would like to query:',
+                choices: [
+                    new Choice('elections', 'Elections'),
+                    new Choice ('demographics', 'Demographics'),
+                    new Choice('households', 'Households'),
+                    new Choice('service_delivery', 'Service Delivery'),
+                    new Choice('economics', 'Economics'),
+                    new Choice('education', 'Education'),
+                    new Choice('children', 'Children'),
+                    new Choice('child_households', 'Child-headed Households')
+                        ],
+                    next: function(choice) {
+                        return {
+                            name: 'states:display-data',
+                            creator_opts : {
+                            choice_name : choice.label,
+                            choice_data : opts.data[choice.value],
+                            location_name : opts.full_name
+                            }
+                        };
+                    }
+            });
+        });
+
+        self.states.add('states:display-data', function(name, opts) {
+            //var choice_data = _.map(opts.choice_data, function(d) {
+            //return new Choice(d); 
+            //});
+            //console.log(choice_data);
+            return new ChoiceState(name, {
+                question: [
+                //data from json file here
+                // '++++++++++++++++++++',
+                opts.location_name,
+                opts.choice_name + ':'
+                //choice_data
+                // '++++++++++++++++++++'
+                ].join('\n'),
+
+                choices: [
+                    new Choice('states:sms', 'SMS details'),
+                    new Choice('states:select-section', 'Query another section'),
+                    new Choice('states:location', 'Change location to query'),
+                    new Choice('states:start', 'Main Menu'),
+                    new Choice('states:end', 'Exit')],
+
+                characters_per_page: 160,
```

Comment:
I don't think `characters_per_page` does anything for `ChoiceState`? But it does for the paginated version which we can switch to later.